Clade.js

```diff
@@ -0,0 +1,682 @@
+			// create one new item per record
+			var record = new marc.record();
+			var newItem = new Zotero.Item();
+			record.leader = ZU.xpathText(rec, "./marc:leader", ns);
+			var fields = ZU.xpath(rec, "./marc:datafield", ns);
+			for (let field of fields) {
+				// go through every datafield (corresponds to a MARC field)
+				var subfields = ZU.xpath(field, "./marc:subfield", ns);
+				var tag = "";
+				for (let subfield of subfields) {
+					// get the subfields and their codes...
+					var code = ZU.xpathText(subfield, "./@code", ns);
+					var sf = ZU.xpathText(subfield, "./text()", ns);
+					// delete non-sorting symbols
+					// e.g. &#152;Das&#156; Adam-Smith-Projekt
+					if (sf) {
+						sf = sf.replace(/[\x80-\x9F]/g, "");
+						// concat all subfields in one datafield, with subfield delimiter and code between them
+						tag = tag + marc.subfieldDelimiter + code + sf;
+					}
+				}
+				record.addField(ZU.xpathText(field, "./@tag", ns), ZU.xpathText(field, "./@ind1", ns) + ZU.xpathText(field, "./@ind2"), tag);
+			}
+
+			record.translate(newItem);
+
+			// Do specific Unimarc postprocessing
+			postProcessMarc(doc, record, newItem);
+
+			// Check for Gallica URL
+			checkGallica(record, newItem);
+
+			newItem.complete();
+		}
+	};
+};
+
+/* Global CladeClass object. */
+var Clade = new CladeClass();
+
+function detectWeb(doc, url) {
+	if (url.search(/\/document\//) != -1) {
+		return getDocumentType(doc);
+	}
+
+	return undefined;
+}
+
+function getDocumentType(doc) {
+	var items = ZU.xpath(doc, '/html/body/main//span[@class="Z3988"][1][@title]');
+
+	if (items.length > 0) {
+		var span = items[0];
+
+		// Get type from dataset
+		if (span.dataset && span.dataset.type !== undefined) {
+			return span.dataset.type;
+		}
+
+		// Get type from item
+		var item = new Zotero.Item;
+		Zotero.Utilities.parseContextObject(span.title, item);
+
+		if (item.itemType) {
+			return item.itemType;
+		}
+	}
+
+	return undefined;
+}
+
+function doWeb(doc, url) {
+	Clade.processMarcUrl(doc, Clade.marcURL(url));
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://bibliotheques-numeriques.defense.gouv.fr/document/e0dd0679-6201-4a8d-a02c-f59957b2b3e6",
+		"items": [
+			{
+				"itemType": "document",
+				"title": "Watteau's soldiers: scenes of military life in eighteenth-century France [exhibition, New York, The Frick Collection, from July 12, 2016 to October 2, 2016]",
+				"creators": [
+					{
+						"firstName": "Aaron",
+						"lastName": "Wile",
+						"creatorType": "contributor"
+					},
+					{
+						"lastName": "Frick collection",
+						"creatorType": "contributor",
+						"fieldMode": true
+					}
+				],
+				"extra": "ill. en noir et en coul., jaquette ill. en coul. 26 cm.",
+				"libraryCatalog": "Clade (https://bibliotheques-numeriques.defense.gouv.fr)",
+				"shortTitle": "Watteau's soldiers",
+				"url": "https://images-na.ssl-images-amazon.com/images/I/51LQ5lqDrzL.jpg",
```

Comment:
This isn't an appropriate value for the URL field, which should be the full, citable online source for a publication. Not clear to me that this site will often provide an appropriate value here.

Reply:
Tests cases has been upated with internal url